server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -5170,6 +5170,8 @@ public void syncSegmentsFromGivenRemoteSegmentStore(
         }
         Map<String, RemoteSegmentStoreDirectory.UploadedSegmentMetadata> uploadedSegments = sourceRemoteDirectory
             .getSegmentsUploadedToRemoteStore();
+        // Checking for existing remote segments in remote
+        boolean hasPreexistingRemoteData = remoteDirectory != null && !remoteDirectory.getSegmentsUploadedToRemoteStore().isEmpty();
```

Comment:
Should we check if the uploaded data represents latest state?

Reply:
check latest state of uploaded data implemented